Vsix/VisualStudioInteraction.cs

```diff
@@ -61,13 +61,28 @@ internal static void Initialize(Cancellation packageCancellation)
             CancelAllToken = packageCancellation.CancelAll;
         }
 
-        public static async Task<string> GetSingleSelectedItemPathOrDefaultAsync()
+        public static async Task<List<string>> GetSelectedItemsPathAsync(Func<string, bool> fileFilter)
         {
             await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(CancelAllToken);
-            var selectedItem = await GetSingleSelectedItemOrDefaultAsync();
-            var itemPath = selectedItem?.ItemPath;
+            const string folderKind = "{6BB5F8EF-4483-11D3-8BCF-00C04F8EC28C}";
+            const string fileKind = "{6BB5F8EE-4483-11D3-8BCF-00C04F8EC28C}";
+
+#pragma warning disable VSTHRD010 // Invoke single-threaded types on Main thread
```

Comment:
I can't quite spot why this pragma was needed - I thought the ThreadHelper line above would mean it wasn't needed, I'll check in VS when I get a chance

Reply:
Accessing anything from the ```ProjectItem``` interface cases this warning.
I guess it's because we do it in lambda? The analyzer thinks it will be lazy-evaluated and executed after we exit UI thread.
You can find a similar code in ```GetSelectedProjectsAsync``` method.

I will change the scope of this pragma so it's a bit smaller and only includes stuff in while loop.